plugins/query-insights/src/test/java/org/opensearch/plugin/insights/QueryInsightsTestUtils.java

```diff
@@ -0,0 +1,170 @@
+        long timestamp = System.currentTimeMillis() + delay;
+        for (int i = 0; i < countOfRecords; ++i) {
+            Map<MetricType, Measurement<Number>> measurements = Map.of(
+                MetricType.LATENCY,
+                new Measurement<>(MetricType.LATENCY.name(), randomLongBetween(1000, 10000)),
+                MetricType.CPU,
+                new Measurement<>(MetricType.CPU.name(), randomDouble()),
+                MetricType.JVM,
+                new Measurement<>(MetricType.JVM.name(), randomDouble())
+            );
+
+            Map<String, Long> phaseLatencyMap = new HashMap<>();
+            int countOfPhases = randomIntBetween(2, 5);
+            for (int j = 0; j < countOfPhases; ++j) {
+                phaseLatencyMap.put(randomAlphaOfLengthBetween(5, 10), randomLong());
+            }
+            Map<Attribute, Object> attributes = new HashMap<>();
+            attributes.put(Attribute.SEARCH_TYPE, SearchType.QUERY_THEN_FETCH.toString().toLowerCase(Locale.ROOT));
+            attributes.put(Attribute.SOURCE, "{\"size\":20}");
+            attributes.put(Attribute.TOTAL_SHARDS, randomIntBetween(1, 100));
+            attributes.put(Attribute.INDICES, randomArray(1, 3, Object[]::new, () -> randomAlphaOfLengthBetween(5, 10)));
+            attributes.put(Attribute.PHASE_LATENCY_MAP, phaseLatencyMap);
+
+            records.add(new SearchQueryRecord(timestamp, measurements, attributes));
+            timestamp += interval;
+        }
+        return records;
+    }
+
+    // public static String createFixSearchQueryRecordString() {
+    //
+    // }
+    public static SearchQueryRecord createFixedSearchQueryRecord() {
+        long timestamp = 1706574180000L;
+        Map<MetricType, Measurement<Number>> measurements = Map.of(MetricType.LATENCY, new Measurement<>(MetricType.LATENCY.name(), 1L));
+
+        Map<String, Long> phaseLatencyMap = new HashMap<>();
+        Map<Attribute, Object> attributes = new HashMap<>();
+        attributes.put(Attribute.SEARCH_TYPE, SearchType.QUERY_THEN_FETCH.toString().toLowerCase(Locale.ROOT));
+
+        return new SearchQueryRecord(timestamp, measurements, attributes);
+    }
+
+    public static void compareJson(ToXContent param1, ToXContent param2) throws IOException {
+        if (param1 == null || param2 == null) {
+            assertNull(param1);
+            assertNull(param2);
+            return;
+        }
+
+        ToXContent.Params params = ToXContent.EMPTY_PARAMS;
+        XContentBuilder param1Builder = jsonBuilder();
+        param1.toXContent(param1Builder, params);
+
+        XContentBuilder param2Builder = jsonBuilder();
+        param2.toXContent(param2Builder, params);
+
+        assertEquals(param1Builder.toString(), param2Builder.toString());
+    }
+
+    @SuppressWarnings("unchecked")
+    public static boolean checkRecordsEquals(List<SearchQueryRecord> records1, List<SearchQueryRecord> records2) {
```

Comment:
you should implement an `equals()` method in `SearchQueryRecord `.

Reply:
If we have implemented equals do we still need this?

Reply:
We still need to compare the attributes for those 2 records. and attributes can be `Map` or `array`, we need to check if they are deeply equal.